webrender/src/render_backend.rs

```diff
@@ -754,6 +757,13 @@ impl RenderBackend {
         // inflight messages, otherwise the scene builder might panic.
         while let Ok(msg) = self.scene_rx.recv() {
             match msg {
+                SceneBuilderResult::FlushComplete(tx) => {
+                    // If somebody's blocked waiting for a flush, how did they
+                    // trigger the RB thread to shut down? This shouldn't happen
+                    // but handle it gracefully anyway.
+                    debug_assert!(false);
```

Comment:
This would be better as ```unreachable!("bug: [reasons]");```

Reply:
unreachable! will panic in release builds, right? I'd rather avoid that in this case because there is a reasonable way to handle it gracefully. That's why I went with the debug_assert instead. This scenario could occur if somebody uses WR with multiple API instances one of which is used to shut down the RB thread while the other is blocked on a flush. I don't believe it's possible with either gecko or servo at the moment, but other WR consumers might want to do that. At the same time, if that scenario does occur we might want to revisit this.